tableauserverclient/models/data_freshness_policy_item.py

```diff
@@ -0,0 +1,210 @@
+            return DataFreshnessPolicyItem.FreshAt(frequency, time, timezone, interval)
+
+    def __init__(self, option: Option):
+        self.option = option
+        self.fresh_every_schedule: Optional[DataFreshnessPolicyItem.FreshEvery] = None
+        self.fresh_at_schedule: Optional[DataFreshnessPolicyItem.FreshAt] = None
+
+    def __str__(self):
+        return '<DataFreshnessPolicy option={_option}>'.format(**vars(self))
+    def __repr__(self):
+        return self.__str__() + "  { " + ", ".join(" % s: % s" % item for item in vars(self).items()) + "}"
+
+    @property
+    def option(self) -> Option:
+        return self._option
+
+    @option.setter
+    @property_is_enum(Option)
+    def option(self, value: Option):
+        self._option = value
+
+    @property
+    def fresh_every_schedule(self) -> Optional[FreshEvery]:
+        return self._fresh_every_schedule
+
+    @fresh_every_schedule.setter
+    def fresh_every_schedule(self, value: fresh_every_schedule):
+        self._fresh_every_schedule = value
+
+    @property
+    def fresh_at_schedule(self) -> Optional[FreshAt]:
+        return self._fresh_at_schedule
+
+    @fresh_at_schedule.setter
+    def fresh_at_schedule(self, value: fresh_at_schedule):
+        self._fresh_at_schedule = value
+
+    @classmethod
+    def from_xml_element(cls, data_freshness_policy_elem, ns):
+
+        option = data_freshness_policy_elem.get("option", None)
+        if option is None:
+            return None
+        data_freshness_policy = DataFreshnessPolicyItem(option)
+
+        fresh_at_schedule = None
+        fresh_every_schedule = None
+        if option == 'FreshAt':
+            fresh_at_schedule_elem = data_freshness_policy_elem.find(".//t:freshAtSchedule", namespaces=ns)
+            fresh_at_schedule = DataFreshnessPolicyItem.FreshAt.from_xml_element(fresh_at_schedule_elem, ns)
+            data_freshness_policy.fresh_at_schedule = fresh_at_schedule
+        elif option == 'FreshEvery':
+            fresh_every_schedule_elem = data_freshness_policy_elem.find(".//t:freshEverySchedule", namespaces=ns)
+            fresh_every_schedule = DataFreshnessPolicyItem.FreshEvery.from_xml_element(fresh_every_schedule_elem)
+            data_freshness_policy.fresh_every_schedule = fresh_every_schedule
+
+        return data_freshness_policy
+
+
+def parse_intervals(intervals_elem, frequency, ns):
+    interval_elems = intervals_elem.findall(".//t:intervals/t:interval", namespaces=ns)
+    interval = []
+    for interval_elem in interval_elems:
+        interval.extend(interval_elem.attrib.items())
+
+    # No intervals expected for Day frequency
+    if frequency == DataFreshnessPolicyItem.FreshAt.Frequency.Day:
+        return None
+
+    if frequency == DataFreshnessPolicyItem.FreshAt.Frequency.Week:
+        interval_values = [(i[1]).title() for i in interval]
+        return parse_week_intervals(interval_values)
+
+    if frequency == DataFreshnessPolicyItem.FreshAt.Frequency.Month:
```

Comment:
Can't this re-use the existing schedule parsing code?

Reply:
I was considering that previously, but turns out there's some differences in how it's implemented - hence the simplest way for us is to have our own interval parser. 
E.g. in schedule, the parser returns MonthlyInterval/WeeklyInterval type +  has different member var, different structure than what we have.